SpatialGDK/Source/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -51,9 +51,34 @@ bool USpatialGameInstance::HasSpatialNetDriver() const
 	return bHasSpatialNetDriver;
 }
 
+void MySpatialBrowse(FWorldContext& WorldContext, FURL URL)
+{
+	bool bIsSpatial = false;
+	UNetDriver* TempNetDriver = nullptr;
+	if (GEngine->CreateNamedNetDriver(WorldContext.PendingNetGame, NAME_PendingNetDriver, NAME_GameNetDriver))
+	{
+		TempNetDriver = GEngine->FindNamedNetDriver(WorldContext.PendingNetGame, NAME_PendingNetDriver);
+		// Setting the PendingNetGame's NetDriver is necessary here because the call to CreateNamedNetDriver above will interfere
+		// with the internals of InitNetDriver and cause the NetDriver not to be initialized, and fail a check().
+		WorldContext.PendingNetGame->NetDriver = TempNetDriver;
+		bIsSpatial = TempNetDriver->IsA(USpatialNetDriver::StaticClass());
+	}
+
+	// Create the proper PendingNetGame depending on what NetDriver we have loaded.
+	// This is required so that we don't break vanilla Unreal networking with SpatialOS switched off.
+	if (bIsSpatial)
+	{
+		WorldContext.PendingNetGame = NewObject<USpatialPendingNetGame>();
+		WorldContext.PendingNetGame->Initialize(URL);
+		// See above comment about setting NetDriver manually here.
+		WorldContext.PendingNetGame->NetDriver = TempNetDriver;
+	}
+}
 
 bool USpatialGameInstance::StartGameInstance_SpatialGDKClient(FString& Error)
 {
+	GetEngine()->SpatialBrowseDelegate = &MySpatialBrowse;
```

Comment:
Do we know that this is early enough to set the delegate? Is it possible that Browse will have already been called at this point? My guess is no, but it would be good to verify.

Reply:
It would appear that the answer is yes.